pkg/translate/messages.go

```diff
@@ -2,20 +2,67 @@ package translate
 
 import (
 	"context"
+	"errors"
 
+	"github.com/expect-digital/translate/pkg/model"
+	"github.com/expect-digital/translate/pkg/repo"
 	pb "github.com/expect-digital/translate/pkg/server/translate/v1"
 	"github.com/expect-digital/translate/pkg/transform"
+	"golang.org/x/exp/slices"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+func (t *TranslateServiceServer) UpdateMessages(
+	ctx context.Context,
+	req *pb.UpdateMessagesRequest,
+) (*pb.UpdateMessagesResponse, error) {
+	var (
+		reqTranslationID = req.GetTranslationId()
+		reqMessages      = req.GetMessages()
+	)
+
+	if len(reqTranslationID) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "missing translation_id")
+	}
+
+	reqMessages.TranslationId = reqTranslationID
+
+	messages, err := transform.MessagesFromProto(reqMessages)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "parsing messages: %s", err)
+	}
+
+	messagesList, err := t.repo.LoadMessages(reqTranslationID)
+
+	switch {
+	case errors.Is(err, repo.ErrNotFound):
+		return nil, status.Errorf(codes.NotFound, "no translations with id '%s'", reqTranslationID)
+	case err != nil:
+		return nil, status.Errorf(codes.Internal, "loading messages: %s", err)
+	}
+
+	if cnt := slices.ContainsFunc(messagesList, func(m model.Messages) bool {
+		return m.Language.String() == messages.Language.String()
+	}); !cnt {
+		return nil, status.Errorf(codes.NotFound, "no such language exists in Database '%s'", messages.Language.String())
+	}
```

Comment:
Here I see two possible solutions:
- First one
language is in the path or as query parameter, to indicate which id and language we want to update. e.g. `PUT /messages/{id}/{language}` or `PUT /messages/{id}?language=...` If no such translation_id or translation_id with that language error is thrown
- Second one
in the path there is only translation_id and language is in body, as our model.messages. In this approach we can change everything except language, that means, if language is something that does not exist in our DB error is thrown.

For now second approach is implemented, but I think first one is better, but need opinions